### PR TITLE
Remove loadClerk

### PR DESCRIPTION
**Pull request by colinclerk**
# Remove loadClerk

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 🌟 New feature (non-breaking change which adds functionality)
- [x] 🔨 Breaking change (fix or feature that would cause existing functionality
- [ ] 📖 Docs change / refactoring / dependency upgradeto change)

## Description

This change removes the need to call `loadClerk` before using the <ClerkProvider> component.  I moved all the loadClerk options to ClerkProvider with the exception of `ssrKey`, which no longer serves a purpose for an unrelated reason.

#### Before this PR

```jsx
loadClerk("auth.foo.com", {
  navigate: (to: string) => { Router.push(to) }
});

const App = ({children}) => {
  return <ClerkProvider>{children}</ClerkProvider>;
}
```

#### After this PR

```jsx
const App = ({children}) => {
  const navigate = (to: string) => { Router.push(to) };
  return <ClerkProvider host="auth.foo.com" navigate={navigate}>{children}</ClerkProvider>;
}
```

### CODE REVIEW

**Comment by colinclerk** [comment]
Note: this requires no additional changes in clerk_js.

The additional changes to clerk_ui are here:
https://github.com/clerkinc/clerk_ui/commit/3abeb235f8753b148481c685cbb8abe236f8c61f

With this PR, I plan to cease work on clerk_js/clerk-react until they are merged into the monorepo.

File: src/index.tsx [hunk 1]
```diff
@@ -315,53 +321,38 @@ class Portal extends React.PureComponent<MountProps, {}> {
...
-    const c = useClerk();
-    if (props.mode && props.mode === 'mounted') {
-      return (
-        <Portal mount={c.mountSignIn} unmount={c.unmountSignIn} props={props} />
-      );
-    } else {
-      return <Modal open={c.openSignIn} close={c.closeSignIn} props={props} />;
-    }
+    return (
+      <Portal
```
Comment (SokratisVidros) [inline src/index.tsx:327]: 👏

File: src/index.tsx [hunk 2]
```diff
@@ -600,31 +584,25 @@ class IsomorphicClerk implements IsomorphicClerkType {
...
-        const ssrDataNode = document.querySelector(
-          `script[data-clerk="${this.ssrKey}"]`
-        );
-        if (ssrDataNode) {
-          this.ssrData = ssrDataNode.innerHTML;
-          const parsedData = JSON.parse(this.ssrData);
-          this.ssrClient = parsedData.client;
-          this.ssrSession = parsedData.session;
-        }
+      const ssrDataNode = document.querySelector(`script[data-clerk="SSR"]`);
```
Comment (SokratisVidros) [inline src/index.tsx:600]: ❓ Is this `<script/>` injected from `clerk.js`

File: src/index.tsx [hunk 3]
```diff
@@ -72,7 +70,15 @@ class ClerkProvider extends React.PureComponent<ContextProps, ContextState> {
     super(props);
     if (props.clerk) {
       this.clerk = props.clerk;
-    } else if (isomorphicClerk) {
+    } else if (props.host) {
```
Comment (SokratisVidros) [inline src/index.tsx:73]: ❓ `host` is the most crucial prop that developers need to pass. Is the term we will use? Currently it's called `Client API` in dashboard.

**Comment by SokratisVidros** [comment]
> Note: this requires no additional changes in clerk_js.
> 
> The additional changes to clerk_ui are here:
> [clerkinc/clerk_ui@3abeb23](https://github.com/clerkinc/clerk_ui/commit/3abeb235f8753b148481c685cbb8abe236f8c61f)
> 
> With this PR, I plan to cease work on clerk_js/clerk-react until they are merged into the monorepo.

Agreed. Let's review them ASAP so that we can start merging the rest.

File: src/index.tsx (hunk 2 above)
Comment (colinclerk) [reply src/index.tsx:600]: It's actually from clerk-react, here: https://github.com/clerkinc/clerk-react/blob/cs/noloadclerk/src/index.tsx#L140-L148

The SSR code is far from complete, so this is a proof-of-concept lingering around.

With SSR, the `Client` object is loaded on the server and passed to the client in this <script> tag so there's no need to wait for `/v1/client` before first paint.

File: src/index.tsx (hunk 3 above)
Comment (colinclerk) [reply src/index.tsx:73]: I don't like it either but haven't thought of anything better yet.

```
// clientAPI feels a little long to me
<ClerkProvider clientAPI="auth.foo.com">

// provider is already used, but maybe we rename ClerkProvider
<ClerkProvider provider="auth.foo.com">

// url could work, but we probably want to add https://
<ClerkProvider url="https://auth.foo.com">
```

File: src/index.tsx (hunk 2 above)
Comment (SokratisVidros) [reply src/index.tsx:600]: 👍

**Review by SokratisVidros — APPROVED** [review]

File: src/index.tsx (hunk 3 above)
Comment (SokratisVidros) [reply src/index.tsx:73]: `instance` maybe?

**Pull request merged**